src/analytics/lambdas/scan-metadata-workflow/store-metadata-into-ddb.ts

```diff
@@ -262,10 +165,105 @@ async function batchWriteIntoDDB(metadataItems: any[]) {
         [ddbTableName]: itemsChunk,
       },
     };
-    await ddbDocClient.send(new BatchWriteCommand(inputPara));
+    try {
+      await ddbDocClient.send(new BatchWriteCommand(inputPara));
+    } catch (error) {
+      // log error and inputPara
+      if (error instanceof Error) {
+        logger.error('Error when batch write into ddb.', error);
+        logger.error('Error when batch write into ddb with inputPara: ', { inputPara });
+      }
+    }
+  }
+}
+
+// limit the size of set to 1000 for events in parameter
+function addSetIntoAnotherSet(sourceSet: Set<string>, inputSet: Set<string>) {
+  for (const element of inputSet) {
+    if (sourceSet.size < 1000) {
+      sourceSet.add(element);
+    }
+  }
+}
+
+/**
+ * Check whether currentMonth is latest month for id and update earlier month to its origin month
+ * @param memoryItemMap memory item map
+ * @param id id
+ * @param currentMonth current month
+ * @param markedLatestMonthMap marked latest month map
+ * @returns
+ * latest: if currentMonth is latest
+ * originMonth: if currentMonth is not latest
+ **/
+async function getAndMarkMonthValue(memoryItemMap: Map<string, any>, id: string, currentMonth: string, markedLatestMonthMap: Map<string, any>) {
+  try {
+    let monthValue;
+    if (!markedLatestMonthMap.has(id)) {
+      monthValue = await checkLatestMonthWithDDB(id, currentMonth, markedLatestMonthMap);
+    } else {
+      monthValue = checkLatestMonthWithMemory(memoryItemMap, id, currentMonth, markedLatestMonthMap);
+    }
+    if (monthValue === 'latest') {
+      // update memory latest month
+      markedLatestMonthMap.set(id, currentMonth);
+    }
+    return monthValue;
+  } catch (error) {
+    if (error instanceof Error) {
```

Comment:
It's not necessary to check the error type.

Reply:
Refactored